app/main/views/suppliers.py

```diff
@@ -408,13 +409,16 @@ def update_supplier_framework_details(supplier_id, framework_slug):
     if not interest_record:
         abort(404, "supplier_id '{}' has not registered interest in {}".format(supplier_id, framework_slug))
 
+    uniform_now = datetime.utcnow()
+
     if 'onFramework' in update_json:
         interest_record.on_framework = update_json['onFramework']
-    if 'agreementReturned' in update_json:
-        if update_json['agreementReturned']:
-            interest_record.agreement_returned_at = datetime.utcnow()
-        else:
-            interest_record.agreement_returned_at = None
+    if update_json.get('agreementReturned'):
+        interest_record.agreement_returned_at = uniform_now
+    if update_json.get('countersigned'):
+        interest_record.countersigned_at = uniform_now
+    if 'signerDetails' in update_json:
+        interest_record.signer_details = update_json["signerDetails"]
```

Comment:
How would an API user "delete" a key from the json then? Would they set the key to json `null` and rely on `purge_nulls_from_data` turning that into a removal? Because we haven't added the necessary validator to the db field, trying to not overthink the field behaviour. (Some json fields have validators doing this, some don't - we guessed that ours shouldn't)

If we're going with this change should we add the json-cleaning validators to the model?